libs/keycloak-admin-client/src/defs/ssoProtocols.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Based upon the keycloak documentation could be found here
+ * https://www.keycloak.org/docs/latest/server_admin/#sso-protocols
+ * or here
+ * https://wjw465150.gitbooks.io/keycloak-documentation/content/server_admin/topics/sso-protocols.html
+ * there are actually three different SSO Protocols named and implemented
+ */
+export const SSOProtocol = {
+  "openid-connect": "openid-connect",
+  saml: "saml",
+  "docker-v2": "docker-v2",
+};
+export type SSOProtocol = typeof SSOProtocol[keyof typeof SSOProtocol];
```

Comment:
this could be simpler right?
```suggestion
export type SSOProtocol = "openid-connect" | "saml" | "docker-v2";
```

Reply:
This is a possibility but, in case we are adding new values to the SSO const we need to add them manually also to the type definition ore else it would differ from each other and lead to a syntax issue. 

May we can add your suggestion as commit above and link the official typescript docu that explains the behavior of this single line. 

In my first days with this line I also thought it is a little bit over complex but we stepped in to many problems that the 'setter' const does not fit at the end to the type validator ... so we replaced the direkt string types by this and based type on configuration.

Reply:
what kind of problems did you run into? we use this everywhere, didn't have any problems. And when the const changes the type needs to change as well, but that is true for your version. WDYT @jonkoops

Reply:
```
 const SSOProtocol = {
    openIDConnect: "openid-connect",
    saml:"saml",
    docker: "docker-v2" 
}
 type SSOProtocol = typeof SSOProtocol[keyof typeof SSOProtocol];

const x: SSOProtocol = "openid-connect" // Not a good idea at all. It is a REST-API value and should be represented by ENUM
console.log(SSOProtocol.openIDConnect) // Should be used becasue it could be ignored if value of openIDConnect will change


type SSOProtocolAlternative = "openid-connect" | "saml" | "docker-v2";

const y:SSOProtocolAlternative = "openid-connect" // Same so fare 
console.log(SSOProtocolAlternative.openid-connect) // ERROR! Could not found
```

On a fluid API or an API on change, it will always cause trouble in case you hard code property-values that are constructed as ENUM values within the REST API in your Client Code. 
In case you are using `SSOProtocolAlternative` your whole code need a refactor on multiple positions (see test cases, I switched from direct using "openeid-connect" to `SSOProtocol.openIDconnect`)  when the API changes.
Ok to be honest, this part of API should be relative stable, but ... 

In case you are using `SSOProtocol.openIDconnect` you just change the value of a property in 1 line, and everything will transpire correctly and because it is defined within the lib you are consuming all user of nom just need an update of package :)

Reply:
No sorry, I will not do that. If you prefer to have it based on string literals, that's entirely your own decision within the community, which has already done a really good job! But from my point of view, besides KIS and Clean Code, the rule is everywhere: Never hardcode values. It may be common use to implement things this way, also because string enums have slipped in only in TS >= 2.4 and until today somehow do not quite fit into the overall concept. 
There are things I don't train myself to avoid after years of practice. :) You now know the problem and a solution. Then do it yourself with pleasure in your own way.